alibi_detect/saving/_sklearn/tests/test_saving_sk.py

```diff
@@ -0,0 +1,32 @@
+from pytest_cases import param_fixture, parametrize, parametrize_with_cases
+
+from alibi_detect.saving.tests.datasets import ContinuousData
+from alibi_detect.saving.tests.models import classifier_model, xgb_classifier_model
+
+from alibi_detect.saving.loading import _load_model_config
+from alibi_detect.saving.saving import _path2str, _save_model_config
+from alibi_detect.saving.schemas import ModelConfig
+
+backend = param_fixture("backend", ['sklearn'])
+
+
+@parametrize_with_cases("data", cases=ContinuousData.data_synthetic_nd, prefix='data_')
+@parametrize('model', [classifier_model, xgb_classifier_model])
+def test_save_model_sk(data, model, tmp_path):
```

Comment:
We arguably could be more specific about what tests we keep in `saving/tests/test_saving.py` and what ones we keep here.

For now, I've adopted moving tests that are very backend specific to the backend saving subpackages (e.g. `_save_model_config` is tested in `saving/_sklearn/tests/test_saving_sk.py` when `model` is a sklearn or xgboost model). However, for the more generic functional tests such as `test_save_ksdrift`, I test them with both `backend`'s in `saving/tests/test_saving.py` to minimise duplication. There are some other tests in `saving/tests/test_saving.py` such as `test_save_optimizer` that we could probably think about moving to the backend-specific testing files in the future...

Also note, our more usual convention would be to test `saving._sklearn.save_model` here, instead of the parent function `saving._save_model_config`. I am testing `_save_model_config` here so that we also test the logic inside, which should call `saving._sklearn.save_model` if it recognises `model` to be a `sklearn.base.BaseEstimator`.